### PR TITLE
migrate to ZIO 2

### PR DESCRIPTION
**Pull request by myazinn**

### CODE REVIEW

File: src/it/scala/zio/container/Layers.scala [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                       .addContactPoints(Seq(address).asJavaCollection)
+                       .withLocalDatacenter("datacenter1")
+                       .withConfigLoader(new DefaultDriverConfigLoader(() => config, false))
+                       .withKeyspace(keyspace)
+        _         <- ensureKeyspaceExists(builder)
+
+        session <- Session.make(builder)
+        _       <- migrateSession(session)
+      } yield session
+    }.mapError(TestFailure.die)
```
Comment (narma) [inline src/it/scala/zio/container/Layers.scala:78]: `mapError` is no longer necessary

File: src/it/scala/zio/container/Layers.scala [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            )
+          )
+          .unless(session.getMetadata.getKeyspace(keyspace).isPresent)
+      _       <- ZIO.fromCompletionStage(session.closeAsync())
+    } yield ()
+
+  val layerCassandra: ZLayer[Scope, TestFailure[Nothing], CassandraContainer] = ZTestContainer.cassandra
+
+  val layerSession: ZLayer[Scope with CassandraContainer, TestFailure[Nothing], Session] =
+    ZLayer.fromZIO {
```
Comment (narma) [inline src/it/scala/zio/container/Layers.scala:62]: same with `ZLayer.scope`

File: src/it/scala/zio/container/Layers.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          .fromCompletionStage(
+            session.executeAsync(
+              s"CREATE KEYSPACE IF NOT EXISTS $keyspace WITH replication = {'class':'SimpleStrategy', 'replication_factor':1};"
+            )
+          )
+          .unless(session.getMetadata.getKeyspace(keyspace).isPresent)
+      _       <- ZIO.fromCompletionStage(session.closeAsync())
+    } yield ()
+
+  val layerCassandra: ZLayer[Scope, TestFailure[Nothing], CassandraContainer] = ZTestContainer.cassandra
```
Comment (narma) [inline src/it/scala/zio/container/Layers.scala:59]: same with ZLayer.scoped

File: src/it/scala/zio/container/ZTestContainer.scala [hunk 4]
```diff
@@ -3,25 +3,25 @@ package zio.container
...
 object ZTestContainer {
 
-  def cassandra: ZLayer[Blocking, TestFailure[Nothing], Has[CassandraContainer]] =
-    managed(CassandraContainer(dockerImageNameOverride = DockerImageName.parse("cassandra:4.0.3")))
-      .mapError(TestFailure.die)
-      .toLayer
+  val cassandra: ZLayer[Scope, TestFailure[Nothing], CassandraContainer] =
+    ZLayer.fromZIO {
+      scoped(CassandraContainer(dockerImageNameOverride = DockerImageName.parse("cassandra:4.0.3")))
+    }.mapError(TestFailure.die)
```
Comment (narma) [inline src/it/scala/zio/container/ZTestContainer.scala:14]: same with `mapError`

File: src/it/scala/zio/container/Layers.scala (hunk 1 above)
Comment (myazinn) [reply src/it/scala/zio/container/Layers.scala]: done

File: src/it/scala/zio/container/Layers.scala (hunk 2 above)
Comment (myazinn) [reply src/it/scala/zio/container/Layers.scala]: done

File: src/it/scala/zio/container/Layers.scala (hunk 3 above)
Comment (myazinn) [reply src/it/scala/zio/container/Layers.scala]: done

**Review by narma — DISMISSED** [review]

**Pull request merged**